vernac/comCheck.mli

```diff
@@ -0,0 +1,51 @@
+(************************************************************************)
+(*         *   The Coq Proof Assistant / The Coq Development Team       *)
+(*  v      *         Copyright INRIA, CNRS and contributors             *)
+(* <O___,, * (see version control and CREDITS file for authors & dates) *)
+(*   \VV/  **************************************************************)
+(*    //   *    This file is distributed under the terms of the         *)
+(*         *     GNU Lesser General Public License Version 2.1          *)
+(*         *     (see LICENSE file for the text of the license)         *)
+(************************************************************************)
+
+type check_result = {
+  env : Environ.env;
+  sigma : Evd.evar_map;
+  judgment : EConstr.unsafe_judgment;
+  evars : Evar.Set.t;
+}
+
+type eval_result = {
+  env : Environ.env;
+  sigma : Evd.evar_map;
+  judgment : EConstr.unsafe_judgment;
+  redfun : Reductionops.reduction_function;
+  rc : Constrexpr.constr_expr;
+}
+
+type global_check_result = {
+  env : Environ.env;
+  sigma : Evd.evar_map;
+  judgment : Safe_typing.judgment;
+}
+
+val check :
+  Environ.env ->
+  Evd.evar_map ->
+  Constrexpr.constr_expr ->
+  check_result * Univ.ContextSet.t
+
+val eval :
+  Environ.env ->
+  Evd.evar_map ->
+  Genredexpr.raw_red_expr ->
+  Constrexpr.constr_expr ->
+  eval_result * Univ.ContextSet.t
```

Comment:
Looks very nice! A couple of further wishlists if you feel like doing an iteration:
- wrapping the result in a record `'a With_contextSet.t` would help having nicer serialization
- Maybe some ocamldocs could be helpful to the reader.